service/mongodb/store.go

```diff
@@ -135,15 +140,28 @@ func (s MongoStore) Create(ctx context.Context, permission service.Permission) (
 		},
 	}
 
-	opts := options.FindOneAndUpdate().SetUpsert(true).SetReturnDocument(options.After)
-	result := collection.FindOneAndUpdate(ctx, filter, update, opts)
-	newPermission := &BSON{}
-	err := result.Decode(newPermission)
+	if override {
+		opts := options.FindOneAndUpdate().SetUpsert(true).SetReturnDocument(options.After)
+		result := collection.FindOneAndUpdate(ctx, filter, update, opts)
+		newPermission := &BSON{}
```

Comment:
rename to `updatedPermission`, and rename `permissionUpdate` to `newPermission`.